src/System.Runtime.Extensions/tests/System/Environment.GetEnvironmentVariable.cs

```diff
@@ -265,6 +267,10 @@ public void EnumerateEnvironmentVariables(EnvironmentVariableTarget target)
             }
         }
 
+        [DllImport("kernel32.dll", EntryPoint = "SetEnvironmentVariableW" , CharSet = CharSet.Unicode, SetLastError = true)]
+        private static extern bool SetEnvironmentVariable(string lpName, string lpValue);
+
+#if !uap
         private static void SetEnvironmentVariableWithPInvoke(string name, string value)
```

Comment:
You've left the SetEnvironmentVariable pinvoke defined for UAP so this test is valid for UAP. You might have to break it into a Unix and non Unix test, the former behind #if !uap